### ISSUE TITLE
Fix S1854 FP: ref assignments in foreach

### ISSUE THREAD

**Issue by shadow-cs** [SonarSource/sonar-dotnet#2303]
### Description

When assigning to `ref` local (obtained via ref return, C# 7 feature), Sonarlint sees it as useless assignment.

### Repro steps

Compiling following snippet produces the S1854 warning.

```csharp
public void Foo()
{
  Span<int> span = new[] { 42 };
  int j = 0;

  foreach (ref var e in span)
    e = j--;
}
```
### Expected behavior

Ne warning to show up.

### Actual behavior

S1854 is produced.

### Known workarounds

Add a suppression.

### Related information

* SonarC# 7.11.0.8083
* .NET Core SDK 2.1.503
